ocs_ci/deployment/provider_client/storage_client_deployment.py

```diff
@@ -0,0 +1,630 @@
+
+# from ocs_ci.utility.retry import retry
+from ocs_ci.ocs.ui.validation_ui import ValidationUI
+from ocs_ci.ocs.ui.base_ui import login_ui, close_browser
+from ocs_ci.ocs.utils import (
+    setup_ceph_toolbox,
+    enable_console_plugin,
+    run_cmd,
+)
+from ocs_ci.utility.utils import (
+    wait_for_machineconfigpool_status,
+)
+from ocs_ci.utility import templating, version
+from ocs_ci.deployment.deployment import Deployment, create_catalog_source
+from ocs_ci.deployment.baremetal import clean_disk
+from ocs_ci.ocs.resources.storage_cluster import (
+    verify_storage_cluster,
+    # check_storage_client_status,
+)
+from ocs_ci.ocs.resources.storage_client import create_storage_client
+from ocs_ci.ocs.resources.catalog_source import CatalogSource
+from ocs_ci.ocs.bucket_utils import check_pv_backingstore_type
+from ocs_ci.ocs.resources import pod
+from ocs_ci.helpers.helpers import (
+    get_all_storageclass_names,
+    verify_block_pool_exists,
+    # verify_cephblockpool_status,
+    # check_phase_of_rados_namespace,
+)
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.helpers.managed_services import (
+    verify_storageclient,
+    # verify_storageclient_storageclass_claims,
+)
+
+
+log = logging.getLogger(__name__)
+
+
+class StorageClientDeployment(object):
+    def __init__(self):
+        print("Initializing webdriver and login to webconsole")
+        # Call a function during initialization
+        self.initial_function()
+
+        print("Intialization")
+        self.validation_ui_obj = ValidationUI()
+        self.ns_obj = ocp.OCP(kind=constants.NAMESPACES)
+        self.ns_obj.new_project(
+            project_name=constants.BM_DEBUG_NODE_NS, policy=constants.PSA_PRIVILEGED
+        )
+        self.ocp_obj = ocp.OCP()
+        self.storage_cluster_obj = ocp.OCP(
+            kind="Storagecluster", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        self.storage_profile_obj = ocp.OCP(
+            kind="Storageprofile", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        self.config_map_obj = ocp.OCP(
+            kind="Configmap", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        self.pod_obj = ocp.OCP(
+            kind="Pod", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        self.scheduler_obj = ocp.OCP(
+            kind=constants.SCHEDULERS_CONFIG,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+        self.sc_obj = ocp.OCP(kind=constants.STORAGECLASS)
+        self.storageclass = "localblock"
+        self.ocp_version = version.get_semantic_ocp_version_from_config()
+        self.ocs_version = version.get_semantic_ocs_version_from_config()
+        self.storage_class_claims = [
+            constants.CEPHBLOCKPOOL_SC,
+            constants.CEPHFILESYSTEM_SC,
+        ]
+        self.ocs_client_operator = defaults.OCS_CLIENT_OPERATOR_NAME
+        self.deployment = Deployment()
```

Comment:
please move heavy functionality from the constructor, it is not considered a good practice